Unify dependency versions

RenFraser opened the pull request:
- Addresses #419.
- Adds the platform project to the build to avoid having non-kotlin files.
- Removed `gradle.properties` now that we don't need to use it anymore.
- Tested by trying to pull in different dependency versions and seeing the configuration step fail.
- It doesn't show it well in the GitHub diff but in the `shared/build.gradle.kts` and `server/build.gradle.kts` files, the dependency declarations have just had their versions truncated.
  - Example `implementation("com.h2database:h2:1.4.200")` becomes `implementation("com.h2database:h2")`. Note the missing `:1.4.200`
- Committed `gradle.properties` to the test additional workspaces since it's used during testing.
  - Removed Gradle tasks that copied the `gradle.properties` to the test additional workspaces.

themkat commented on server/build.gradle.kts:
```diff
@@ -76,20 +78,10 @@ tasks.startScripts {
 tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().all {
     kotlinOptions {
-        jvmTarget = javaVersion
+        jvmTarget = "11"
```
This change reintroduces the issue that was fixed for the Java 17 build in #426 . Quick suggestion to fix it: 
```kotlin
val javaVersion = if (project.hasProperty("javaVersion")) project.property("javaVersion").toString() else "11"
```

Then set `jvmTarget = javaVersion`. This will at least make the Java 17 workflow work without deprecation warnings. 

I see that we specify Java 11 in the shared module as well, but unsure on why that doesn't cause any warnings. Have to have a second look at that.

RenFraser replied:
Ok no problem. I've added the ability to override it as you've suggested but with a slight tweak. :)

themkat commented on build.gradle.kts:
```diff
@@ -1,6 +1,3 @@
-// We could use a buildSrc script for this.
-// See: https://github.com/gradle/kotlin-dsl-samples/issues/802
 plugins {
     kotlin("jvm") version "1.8.10"
```
We are still hardcoding 1.8.10 here? Any way we can use the version from the platform project to avoid having this duplicate?

RenFraser replied:
Good point. It's not explicit in my initial revision but if you try to use another version that's different to what's constrained in the platform project, it'll complain stating the the lib is not found. I've refactored the plugins with their versions into a centralised place to live in `settings.gradle.kts` with a comment to make it obvious on how it all works. This will make it easier to see what versions we use.

fwcd commented on gradle/platform/build.gradle.kts:
```diff
@@ -0,0 +1,42 @@
+    id("java-platform")
+}
+
+group = "com.kotlin-language-server"
```
This should ideally match up with the package structure or at least an actual domain. Since the packages live in `org.javacs.kt` (as inherited by the original language server repo), we could use that with the disadvantage of it not being very descriptive.

As an alternative, I would propose

```suggestion
group = "dev.fwcd.kotlin-language-server"
```

Whether we would want to move the actual package structure to that too would be a topic for future debate.

RenFraser replied:
Sure thing, I've made the adjustment in my next revision!

RenFraser commented:
Thanks for the feedback, @fwcd and @themkat. I've pushed a squashed commit with the following changes:

- Added the kotlin jvm plugin with appropriate version to the platform project dependency constraints.
- Renamed the platform project group.
- Added centralised plugin versions in the root `settings.gradle.kts` with supporting comment on what it does and how to add plugins.
- Added logic to use the project property override for `javaVersion` if available and fallback to a default value of `"11"`
- Rebased on top of the latest main branch.

themkat commented:
@RenFraser , if you rebase this branch with main, we can (probably) see that all the pipelines and Detekt are successful 🙂 Always good to check all of those going forward. (you would probably have to do it anyway because of the conflicts. Sorry for nagging, just my enthusiasm for seeing green builds with new stuff 😆  )

RenFraser commented:
Yeah, good idea. Let's resolve those conflicts and see all of our checks passing :) I'll rebase now

RenFraser commented:
Ok that looks good now @themkat :)

fwcd commented:
Hm, I think this broke the `release_version.py` script which relied on parsing and updating the `projectVersion` in `gradle.properties`...

themkat commented:
> Hm, I think this broke the `release_version.py` script which relied on parsing and updating the `projectVersion` in `gradle.properties`...

Is that script even used anymore? Can't see it used in Github Actions

fwcd commented:
Yeah, it's meant for interactive usage

themkat commented:
> Yeah, it's meant for interactive usage

We can probably get it working again by reading `server/build.gradle.kts` instead of `gradle.properties`. The version is in that file now.
